### ISSUE TITLE
Rebamp of logging (both system and viewer)

### ISSUE THREAD

**Issue by francipvb** [nvaccess/nvda#9977]
### Is your feature request related to a problem? Please describe.

This is mainly related to debugging NVDA internal or external code when log level is set to debug or higher. The fact is that it is to difficult to find a particular debug or error message and its information when you write a line to report a variable or something via the logging subsystem.

### Describe the solution you'd like

I think that it should not be difficult, but instead of the current plain text NVDA writes to the buffer, we should add another adapter to save in a structured data format, as well another to report (in special conditions) to a server.

Then we should adapt the log viewer with a list box or something to represent log messages, as well filter controls to filter log records by date and time (we can save a big number of records). By default we should filter log records by the currently running instance and exclude all other entries, but a control should be added to avoid this as well.

Yes, NVDA should be able to export all current visible entries or all records for this running instance or for the filters anyone wants in XML, JSON or any structured data language as well to plain text file.

I thought about a sqlite database as a primary record store.

### Describe alternatives you've considered

None considered.

### Additional context

In these days, almost the majority of applications logs in structured data formats to handle telemetry or debugging at production time. I think that this will enhance NVDA's debugging capabilities and will able developers to work in more specific issues.

**Comment by leonardder** [comment]
For @babbagecom products, we log in csv, which is indeed easier to parse.

I recall @bramd had interesting ideas about it, including an online service that could monitor occurences of errors, warnings and freezes, though I forgot the name of it.

**Comment by bramd** [comment]
@leonardder I was talking about Sentry[0]. I use that to collect errors from my web applications. Could be used for NVDA as well, but I would be a bit concerned about logging sensitive info. I find it very powerful to see when an error is first logged/introduced and how many occurrences there are. For my web apps, we add the user as a tag, so I can see which users are affected and how many.

[0]: https://sentry.io/

**Comment by josephsl** [comment]
Hi,

Five years later...

@LeonardDeR, any thoughts?

Thanks.
